tokenizers/src/tokenizer/mod.rs

```diff
@@ -595,9 +599,21 @@ where
     }
 
     /// Set the truncation parameters
-    pub fn with_truncation(&mut self, trunc: Option<TruncationParams>) -> &mut Self {
+    ///
+    /// Fails if `stride` is too high relative to `max_length` and `post_processor.added_tokens()`
+    pub fn with_truncation(&mut self, trunc: Option<TruncationParams>) -> Result<&mut Self> {
+        if let Some(trunc_params) = &trunc {
+            let n_added_tokens = self.get_n_added_tokens(false);
```

Comment:
Why `false`  ? It depends.

Reply:
I assumed that fewer tokens were added when `is_pair=false` (I'm not actually sure if this is a good assumption). So, when `with_truncation()` is called, we check if there is any possible case where this is a valid combination of `max_length` and `stride` for this tokenizer, and fail if it isn't.

Another reason why I kept the original `assert!()` from the inference part of the code is that it's possible that any additional special token(s) when `is_pair=true` will push it over the edge and make the `stride` too large for the `max_length`

Reply:
@Narsil can you confirm whether my assumption in the above comment is a good one (i.e. is `added_tokens(false)` always less than `added_tokens(true)`?)